topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/MultiMap.java

```diff
@@ -19,60 +19,54 @@
 package org.apache.aries.rsa.topologymanager.importer;
 
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
 
 /**
- * Minimal implementation of a synchronized map
+ * Minimal implementation of a thread-safe map where each key can have multiple values.
+ *
+ * @param <K> the key type
+ * @param <V> the value type
  */
-public class MultiMap<T> {
+public class MultiMap<K, V> {
 
-    private Map<String, Set<T>> map;
+    private Map<K, Set<V>> map;
```

Comment:
I would use the type ConcurrentMap in order to communicate that not any Map implementation will do.

Reply:
Not any ConcurrentMap implementation might do either... some things (like atomicity, iteration safety etc.) are implementation-dependent. I think in general it's best to declare the most abstract interface that's actually being used, though I don't mind too much.